pkg/transform/engine/engine_javascript.go

```diff
@@ -211,12 +227,19 @@ func (e *JSEngine) MakeFunction(funcName string) transform.TransformationFunctio
 			// encode
 			encoded, err := json.Marshal(protoData)
 			if err != nil {
-				message.SetError(fmt.Errorf("error encoding message data"))
+				message.SetError(&models.TransformationError{
+					SafeMessage: "error encoding message data",
+					Err:         err,
```

Comment:
Probably we should wrap this error with the above string as well. Not so useful to the customer but for us, it'll help identify where an obscure error came from.